Backend/Remora.Discord.API/API/Formatting/Markdown.cs

```diff
@@ -0,0 +1,119 @@
+        /// A markdown-formatted italicised string.
+        /// </returns>
+        public static string Italicise(string text) => $"*{text}*";
+
+        /// <summary>
+        /// Formats a string to use Markdown Underlined formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted underlined string.
+        /// </returns>
+        public static string Underline(string text) => $"__{text}__";
+
+        /// <summary>
+        /// Formats a string to use Markdown Strikethrough formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted strikethrough string.
+        /// </returns>
+        public static string Strikethrough(string text) => $"~~{text}~~";
+
+        /// <summary>
+        /// Formats a string to use Markdown Spoiler formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted spoilered string.
+        /// </returns>
+        public static string Spoiler(string text) => $"||{text}||";
+
+        /// <summary>
+        /// Formats a string to use Markdown Inline Code formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted inline code string.
+        /// </returns>
+        public static string InlineCode(string text) => $"`{text}`";
+
+        /// <summary>
+        /// Formats a string to use Markdown Block Code formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted block code string.
+        /// </returns>
+        public static string BlockCode(string text) => $"```\n{text}\n```";
+
+        /// <summary>
+        /// Formats a string to use Markdown Block Code formatting with a specified
+        /// language for syntax highlighting.
+        /// </summary>
+        /// <remarks>
+        /// Discord supports the following languages: <see href="https://github.com/highlightjs/highlight.js/tree/main/src/languages"/>.
+        /// </remarks>
+        /// <param name="text">The input text to format.</param>
+        /// <param name="language">The language.</param>
+        /// <returns>
+        /// A markdown-formatted block code string.
+        /// </returns>
+        public static string BlockCode(string text, string language) => $"```{language}\n{text}\n```";
+
+        /// <summary>
+        /// Formats a string to use Markdown Quote Block formatting.
+        /// </summary>
+        /// <param name="text">The input text to format.</param>
+        /// <returns>
+        /// A markdown-formatted block quote string.
+        /// </returns>
+        public static string BlockQuote(string text) => $">>> {text}";
```

Comment:
Can an ``BlockList`` method be added as well?

It could possibly be something like so:

```cs
public static string BlockList(string text, string mark = "-") => $" {mark} {text}\n";
```
Where mark can be: ``-``, a number (for number based lists that are supported in markdown), an ``*`` for lists with the dots, etc.

Also I think this line in ``BlockQuote`` can be:

```suggestion
        public static string BlockQuote(string text) => $">>> {text}\n";
```

Reply:
Discord doesn't recognize lists in its markdown, so that would be out of scope.